### PR TITLE
Upgrade all Jackson dependencies to 2.12.1 [run-systemtest]

### PR DESCRIPTION
**Pull request by bjorncs**
Replaces https://github.com/vespa-engine/vespa/pull/14949.

### CODE REVIEW

**Comment by bjorncs** [comment]
@baldersheim please review

FYI @gjoranv

File: cloud-tenant-base-dependencies-enforcer/pom.xml [hunk 1]
```diff
@@ -94,6 +95,8 @@
                                         <include>com.sun.xml.bind:jaxb-core:[${jaxb.version}]:jar:provided</include>
                                         <include>com.sun.xml.bind:jaxb-impl:[${jaxb.version}]:jar:provided</include>
                                         <include>commons-logging:commons-logging:[1.1.3]:jar:provided</include>
+                                        <include>jakarta.activation:jakarta.activation-api:[1.2.1]:jar:provided</include>
+                                        <include>jakarta.xml.bind:jakarta.xml.bind-api:[2.3.2]:jar:provided</include>
```
Comment (baldersheim) [inline cloud-tenant-base-dependencies-enforcer/pom.xml:99]: Intentional ? I do not see any use of this package.

**Review by baldersheim — APPROVED** [review]

File: cloud-tenant-base-dependencies-enforcer/pom.xml [hunk 2]
```diff
@@ -41,8 +41,8 @@
 
         <hk2.version>2.5.0-b32</hk2.version>
         <hk2.osgi-resource-locator.version>1.0.1</hk2.osgi-resource-locator.version>
-        <jackson2.version>2.8.11</jackson2.version>
-        <jackson-databind.version>${jackson2.version}.6</jackson-databind.version>
+        <jackson2.version>2.12.1</jackson2.version>
```
[the remark this answers is not in the document]
Comment (bjorncs) [reply cloud-tenant-base-dependencies-enforcer/pom.xml:44]: Old bundles built against 2.8.11 will be able to load in a container providing 2.12.1. The newer version should be backward compatible, although some older APIs are now deprecated (build fails if `Xlint:deprecation` + `Wall` are enabled).

File: cloud-tenant-base-dependencies-enforcer/pom.xml (hunk 1 above)
Comment (bjorncs) [reply cloud-tenant-base-dependencies-enforcer/pom.xml:99]: These are new transitive dependencies from `vespa_jersey2` (from some updated jersey-jackson integration artifact).

File: cloud-tenant-base-dependencies-enforcer/pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (baldersheim) [reply cloud-tenant-base-dependencies-enforcer/pom.xml:44]: K

File: cloud-tenant-base-dependencies-enforcer/pom.xml (hunk 1 above)
Comment (baldersheim) [reply cloud-tenant-base-dependencies-enforcer/pom.xml:99]: K

**Comment by baldersheim** [comment]
Merge at will

File: vespa-http-client/src/main/java/com/yahoo/vespa/http/client/core/communication/ClusterConnection.java [hunk 3]
```diff
@@ -102,6 +102,13 @@ public ClusterConnection(OperationProcessor operationProcessor,
         }
     }
 
+    private static ObjectMapper createMapper() {
```
Comment (gjoranv) [inline vespa-http-client/src/main/java/com/yahoo/vespa/http/client/core/communication/ClusterConnection.java:105]: Duplicated 3 times. A candidate for vespajlib (it already depends on a few jackson artifacts)?

**Pull request merged**